### PR TITLE
Unit tests create

### PR DESCRIPTION
**Pull request by tumf87**
CRUD Panel create trait unit tests.

### CODE REVIEW

File: tests/Unit/CrudPanel/BaseDBCrudPanelTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+use Backpack\CRUD\CrudPanel;
+use Orchestra\Database\ConsoleServiceProvider;
+
+abstract class BaseDBCrudPanelTest extends BaseCrudPanelTest
+{
+    /**
+     * @var CrudPanel
+     */
+    protected $crudPanel;
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/BaseDBCrudPanelTest.php]: Technically this and the below both come in via the base class don't they?

File: tests/Unit/CrudPanel/CrudPanelCreateTest.php [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+        $entry = Article::find(1);
+        $this->crudPanel->syncPivot($entry, $inputData);
+    }
+
+    private function assertEntryEquals($expected, $actual)
+    {
+        foreach ($expected as $key => $value) {
+            if (is_array($value)) {
+                $this->assertEquals(count($value), $actual->{$key}->count());
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/CrudPanelCreateTest.php]: Wouldn't this mean any random array of values with the same number of keys are "equal":

```
$a = [1, 2, 3];
$b = ['a', 'b', 'c'];

# Are they equal?
```

File: tests/Unit/CrudPanel/CrudPanelCreateTest.php [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+            } else {
+                $this->assertEquals($value, $actual->{$key});
+            }
+        }
+
+        $this->assertNotNull($actual->created_at);
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/CrudPanelCreateTest.php]: What do these have with equality?

Actually is this a badly named assertion?

File: tests/Unit/CrudPanel/CrudPanelFakeFieldsTest.php [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    public function testCompactFakeFieldsFromRequestWithNoFakes()
+    {
+        $compactedFakeFields = $this->crudPanel->compactFakeFields($this->noFakeFieldsInputData);
+
+        $this->assertEquals($this->noFakeFieldsInputData, $compactedFakeFields);
+    }
+
+    public function testCompactFakeFieldsFromUnknownForm()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/CrudPanelFakeFieldsTest.php]: If the method *should* throw an exceptoin, but it doesn't, and the test correctly tests that, how is it that the test is incomplete?

(i.e. the test is complete and the thing it's testing has a bug - unless I misunderstand the below comment).

Comment (tabacitu) [reply tests/Unit/CrudPanel/CrudPanelFakeFieldsTest.php]: Haha - guess the naming is a bit off. The test is correctly implemented, but the feature isn't, I guess :-)

**Pull request merged**